examples/intervention_approvals/approvals.py

```diff
@@ -0,0 +1,335 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
+from rich.panel import Panel
+from rich.console import Console, Group, RenderableType
+from rich.markdown import Markdown
+from rich.syntax import Syntax
+from rich.prompt import Confirm, Prompt
+from inspect_ai.util import input_screen
+from inspect_ai.solver._task_state import sample_state
+from inspect_ai.solver import TaskState
+from inspect_ai.model import ModelOutput
+from inspect_ai.tool import ToolCall
+import sys
+import shlex
+
+class ApprovalDecision(Enum):
+    """
+    Enum representing possible decisions for an approval request.
+    """
+    APPROVE = "approve"
+    REJECT = "reject"
+    ESCALATE = "escalate"
+    TERMINATE = "terminate"
+
+class Approver(ABC):
```

Comment:
For "single method" classes like these we use `Protocol` (with a `__call__` method) throughout Inspect, as a `Protocol` can be implemented either as a class or a free function. See `Solver`, `Scorer`, etc. One important reason we do this is that everything in Inspect should be stateless (as all of the instances of classes get used many times over for different samples, etc.) and a free function is much more stateless in implementation by default (essentially just trying to take away a foot gun).